File: src/main/java/infobip/test/shortener/url/OpenUrlCommand.java
package infobip.test.shortener.url;

import infobip.test.shortener.model.Url;
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier;
import org.immutables.value.Value;

@Value.Immutable
public interface OpenUrlCommand {

    @TargetAggregateIdentifier
    String getPath();

    Url getUrl();

}

File: src/main/java/infobip/test/shortener/url/OpenUrlCommand.kt
package infobip.test.shortener.url

import infobip.test.shortener.model.Url
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier

data class OpenUrlCommand(@TargetAggregateIdentifier val path: String? = null,
                          val url: Url? = null)